src/cuga/backend/cuga_graph/utils/token_counter.py

```
@@ -289,6 +289,72 @@
+
+    if model is not None:
+        try:
+            existing = getattr(model, "profile", None)
+            if not isinstance(existing, Mapping) or existing.get("max_input_tokens") != context_size:
+                model.profile = {"max_input_tokens": context_size}
+                logger.debug(
+                    "Set model profile: max_input_tokens={} for {}",
+                    context_size,
+                    resolved_name or model_name or "unknown",
+                )
+        except Exception as exc:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)